src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -773,269 +773,295 @@ static HtmlEntities()
+                    [ToUInt64Key("ang")] = '\x2220',
+                    [ToUInt64Key("and")] = '\x2227',
+                    [ToUInt64Key("or")] = '\x2228',
+                    [ToUInt64Key("cap")] = '\x2229',
+                    [ToUInt64Key("cup")] = '\x222a',
+                    [ToUInt64Key("int")] = '\x222b',
+                    [ToUInt64Key("there4")] = '\x2234',
+                    [ToUInt64Key("sim")] = '\x223c',
+                    [ToUInt64Key("cong")] = '\x2245',
+                    [ToUInt64Key("asymp")] = '\x2248',
+                    [ToUInt64Key("ne")] = '\x2260',
+                    [ToUInt64Key("equiv")] = '\x2261',
+                    [ToUInt64Key("le")] = '\x2264',
+                    [ToUInt64Key("ge")] = '\x2265',
+                    [ToUInt64Key("sub")] = '\x2282',
+                    [ToUInt64Key("sup")] = '\x2283',
+                    [ToUInt64Key("nsub")] = '\x2284',
+                    [ToUInt64Key("sube")] = '\x2286',
+                    [ToUInt64Key("supe")] = '\x2287',
+                    [ToUInt64Key("oplus")] = '\x2295',
+                    [ToUInt64Key("otimes")] = '\x2297',
+                    [ToUInt64Key("perp")] = '\x22a5',
+                    [ToUInt64Key("sdot")] = '\x22c5',
+                    [ToUInt64Key("lceil")] = '\x2308',
+                    [ToUInt64Key("rceil")] = '\x2309',
+                    [ToUInt64Key("lfloor")] = '\x230a',
+                    [ToUInt64Key("rfloor")] = '\x230b',
+                    [ToUInt64Key("lang")] = '\x2329',
+                    [ToUInt64Key("rang")] = '\x232a',
+                    [ToUInt64Key("loz")] = '\x25ca',
+                    [ToUInt64Key("spades")] = '\x2660',
+                    [ToUInt64Key("clubs")] = '\x2663',
+                    [ToUInt64Key("hearts")] = '\x2665',
+                    [ToUInt64Key("diams")] = '\x2666',
                 };
 
-            public static char Lookup(string entity)
+            public static char Lookup(ReadOnlySpan<char> entity)
             {
-                char theChar;
-                s_lookupTable.TryGetValue(entity, out theChar);
-                return theChar;
+                // To avoid an allocation, keys of type "ulong" are used in the lookup table.
+                // Since all entity strings comprise 8 characters or less and are ASCII-only, they "fit" into an ulong (8 bytes).
+                if (entity.Length <= 8)
+                {
+                    s_lookupTable.TryGetValue(ToUInt64Key(entity), out char result);
+                    return result;
+                }
+                else
+                {
+                    // Currently, there are no entities that are longer than 8 characters.
+                    return default;
+                } 
+            }
+
+            private static ulong ToUInt64Key(ReadOnlySpan<char> entity)
+            {
+                // The ulong key is the reversed single-byte character representation of the actual entity string.
```

Comment:
It'd be good to assert here that the length is <= 8. If that ever doesn't hold, this in combination with the lookup table could give incorrect answers.